lib/User_LDAP.php

```diff
@@ -182,11 +182,18 @@ public function getUsers($search = '', $limit = 10, $offset = 0) {
 			$filter,
 			$this->access->userManager->getAttributes(true),
 			$limit, $offset);
-		$ldap_users = $this->access->ownCloudUserNames($ldap_users);
-		\OCP\Util::writeLog('user_ldap', 'getUsers: '.count($ldap_users). ' Users found', \OCP\Util::DEBUG);
+		$ldap_users2 = $this->access->ownCloudUserNames($ldap_users);
+		\OCP\Util::writeLog('user_ldap', 'getUsers: '.count($ldap_users2). ' Users found', \OCP\Util::DEBUG);
+
+		// sanity check
+		$ldap_users_count = count($ldap_users);
+		$ldap_users2_count = count($ldap_users2);
+		if ($ldap_users_count !== $ldap_users2_count) {
+			\OCP\Util::writeLog('user_ldap', "getUsers: fetched $ldap_users_count users but returned $ldap_users2_count", \OCP\Util::ERROR);
+		}
 
-		$this->access->connection->writeToCache($cachekey, $ldap_users);
-		return $ldap_users;
+		$this->access->connection->writeToCache($cachekey, $ldap_users2);
+		return $ldap_users2;
```

Comment:
It would be nice to find a way to both:
1) return just the "good" subset of users (like here); and
2) return the count of all users in this "batch" $ldap_users_count (or a boolean to say if $limit users were found)
So that the caller can know when this may not be the last "batch".

Reply:
with the account table getUsers is only used during sync and during login via saml / shibboleth. for the sync case the OutOfBoundsExceptions are a better way to notify core of preblems with accounts. For the saml login this also works, so I'll close this PR.